src/main/java/org/pac4j/jax/rs/filters/ApplicationLogoutFilter.java

```diff
@@ -22,23 +20,32 @@
 @Priority(Priorities.AUTHORIZATION)
 public class ApplicationLogoutFilter extends AbstractFilter {
 
-    private ApplicationLogoutLogic<Object, JaxRsContext> applicationLogoutLogic = new DefaultApplicationLogoutLogic<>();
+    private static final JaxRsApplicationLogoutLogic<JaxRsContext> DEFAULT_LOGIC = new JaxRsApplicationLogoutLogic<>();
+
+    private ApplicationLogoutLogic<Object, JaxRsContext> applicationLogoutLogic;
 
     private String defaultUrl;
 
     private String logoutUrlPattern;
 
     public ApplicationLogoutFilter(Providers providers, Config config) {
         super(providers, config);
-        ((DefaultApplicationLogoutLogic<Object, JaxRsContext>) applicationLogoutLogic)
-                .setProfileManagerFactory(c -> new JaxRsProfileManager(c));
     }
 
     @Override
     protected void filter(JaxRsContext context) throws IOException {
-        CommonHelper.assertNotNull("applicationLogoutLogic", applicationLogoutLogic);
 
-        applicationLogoutLogic.perform(context, config, adapter(), context.getAbsolutePath(defaultUrl, false),
+        ApplicationLogoutLogic<Object, JaxRsContext> apl;
+
+        if (applicationLogoutLogic != null) {
```

Comment:
I understand, but isn't it too much? Customizing logics should be an edge use case. I would not allow that at first and wait for real use cases to go that far. But you can keep it as is if you prefer.